influxdb_client/client/bucket_api.py

```diff
@@ -117,3 +118,32 @@ def find_buckets(self, **kwargs):
         :return: Buckets
         """
         return self._buckets_service.get_buckets(**kwargs)
+
+    def find_buckets_iter(self, **kwargs):
+        """Iterate over all buckets with pagination.
+
+        :key str name: Only returns buckets with the specified name
+        :key str org: The organization name.
+        :key str org_id: The organization ID.
+        :key str after: The last resource ID from which to seek from (but not including).
+        :key int limit: the maximum number of buckets in one page
+        :return: Buckets iterator
+        """
+
+        def get_next_page(page: _Page):
+            return self._find_buckets_next_page(page, **kwargs)
+
+        return iter(_PageIterator(_Page.initial(kwargs.get('after')), get_next_page))
+
+    def _find_buckets_next_page(self, page: _Page, **kwargs):
+        if not page.has_next:
+            return _Page.empty()
+
+        kw_args = {**kwargs, 'after': page.next_after} if page.next_after is not None else kwargs
+        response = self._buckets_service.get_buckets(**kw_args)
+
+        buckets = response.buckets
+        has_next = response.links.next is not None
+        last_id = buckets[-1].id if buckets else None
+
+        return _Page(buckets, has_next, last_id)
```

Comment:
Please consider refactoring this code to reduce duplication with `influxdb_client.client.tasks_api.TasksApi._find_tasks_next_page`. What do you think about adding a new helper function into `_Page`, perhaps named `_find_next_page(...)`? This could help streamline our codebase.

Reply:
Hello, thanks for your feedback. i've considered it, but a page's content field is resource-type dependent:
eg: for tasks: `response.tasks`, for buckets: `response.buckets`

I could still refactor all this to `find_iter`, so that it'd be:
- in `BucketsAPI`:
```python
 def find_buckets_iter(self, **kwargs):
        return  _Paginated(self._buckets_service.get_buckets, lambda response: response.buckets).find_iter(**kwargs)
```
- in `TasksAPI`:
```python
 def find_tasks_iter(self, **kwargs):
        return  _Paginated(self._tasks_service.get_tasks, lambda response: response.tasks).find_iter(**kwargs)
```

what do you think ?

Reply:
Please yes, it looks like good solution for future maintenance.

Reply:
ok, will do

Reply:
Thanks